nixos/modules/services/networking/6tunnel.nix

```diff
@@ -0,0 +1,111 @@
+{ lib, config, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services._6tunnel;
+
+in {
+  options.services._6tunnel = {
+
+    tunnels = mkOption {
+      default = [];
+      description = "Which tunnels to create.";
+      type = with types; listOf (submodule {
+        options = {
+          port = mkOption {
+            description = "Local port to open.";
+            type = types.port;
+          };
+          host = mkOption {
+            description = "Remote host to tunnel to.";
+            type = types.str;
+          };
+          remotePort = mkOption {
+            default = null;
+            description = "Port on remote host. If null (default), it uses the the local port.";
+            type = with types; nullOr port;
+          };
+          restart = mkOption {
+            default = null;
+            description = ''
+              Whether to restart the tunnel periodically, which is useful for non-static IP addresses.
+              The format is described in SYSTEMD.TIME(7).
+
+              If null (default), the tunnel won't restart periodically.
+            '';
+            type = with types; nullOr str;
+          };
+        };
+      });
+      example = [
+        { port = "8080"; host = "192.168.0.2"; remotePort = 80; }
+        { port = "443"; host = "www.example.com"; }
+        { port = "8053"; host = "dynamicip.example.com"; restart = "1h"; }
+      ];
+    };
+
+    openFirewall = mkOption {
+      default = true;
+      description = "Whether to open the needed firewall ports.";
+      type = types.bool;
+    };
+
+  };
+
+  config = mkIf (cfg.tunnels != {}) {
+
+    networking.firewall.allowedTCPPorts = mkIf (cfg.openFirewall) (map (tunnel: tunnel.port) cfg.tunnels);
+    networking.firewall.allowedUDPPorts = mkIf (cfg.openFirewall) (map (tunnel: tunnel.port) cfg.tunnels);
+
+    systemd.services = listToAttrs (map (tunnel:
+      nameValuePair "6tunnel-${toString tunnel.port}" {
+        description = "6tunnel from ${toString tunnel.port} to ${tunnel.host}${if (tunnel.remotePort != null) then ":${toString tunnel.remotePort}" else ""}";
+        after = [ "network.target" ];
+        wantedBy = [ "multi-user.target" ];
+        serviceConfig = {
+          ExecStart = ''
+            ${pkgs._6tunnel}/bin/6tunnel -d ${toString tunnel.port} "${tunnel.host}" ${optionalString (tunnel.remotePort != null) (toString tunnel.remotePort)}
+          '';
+          Restart = "always";
+          RuntimeMaxSec = mkIf (tunnel.restart != null) tunnel.restart;
+          # Hardening
+          DynamicUser = true;
+          Group = "nogroup";
+          AmbientCapabilities = "CAP_NET_BIND_SERVICE";
+          ProtectSystem = "strict";
+          ProtectHome = true;
+          PrivateTmp = true;
+          PrivateDevices = true;
+          PrivateNetwork = false; # needs full network access
```

Comment:
```suggestion
```
That's the default.

Reply:
I put that in there, because I've seen it in other modules as to show that I've evaluated every single option in the sandboxing part of the manual, but I can remove it if you like.

Reply:
Okay, totally up to you as maintainer.

Reply:
OK, then I'll leave it so I have a sort of "template" for future services.